addons/mail/models/mail_channel.py

```diff
@@ -239,6 +239,10 @@ def _unlink_except_all_employee_channel(self):
             raise UserError(_('You cannot delete those groups, as the Whole Company group is required by other modules.'))
 
     def write(self, vals):
+        if 'channel_type' in vals:
+            for channel in self:
+                if channel.channel_type:
+                    raise UserError(_('Cannot change the channel type of \'%s\' because it is already been set.'), channel.name)
```

Comment:
I would apply the same principle TDE suggested you on the other PR: build the list of all the "failing" channels, and them list them in the error message.

Note: to actually detect change, you need to compare the given `channel_type` to the current channel type of each channel. It shouldn't prevent from giving to write the same value it already had.